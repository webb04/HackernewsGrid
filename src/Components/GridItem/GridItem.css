.GridItem {
    display: block;
    position: absolute;
    width: 200px;
    height: 200px;
    margin: 5px;
    z-index: 1;
    background: #fff;
    overflow: hidden;
    padding: 20px;
    animation: shadow 3s linear forwards;
}

@media only screen and (max-width: 540px) {
    .GridItem {
        width: calc(100% - 40px);
    }
}

@keyframes shadow {
    0% { 
        box-shadow: 0 0 0 transparent; 
    } 
    100% {
        box-shadow: 0 2px 10px rgba(0,0,0,.15); 
    }
}

.GridItem:hover {
    cursor: pointer;
}

.GridItem.selected {
    height: 450px;
    width: 450px;
    z-index: 100;
    background: #FBDDB8;
}

.score {
    display: inline-block;
    width: 50px;
    float: left;
    font-size: 18px;
    font-weight: 200;
}

.GridItem.selected .details {
    width: 350px;
}

.details {
    display: inline-block;
    width: 150px;
    overflow: hidden;
}

.by {
    padding: 10px 0px;
    font-size: 14px;
    color: #404040;
}

.title {
    font-size: 18px;
}