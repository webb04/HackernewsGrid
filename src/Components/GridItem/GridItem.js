import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import './GridItem.css';

class GridItem extends Component {
    render() {
        const { by, title, score, selected, gridKey, zIndex, onSelect } = this.props;
        const GridItemClasses = classnames({
            GridItem: true,
            selected
        });
        return (
            <div style={{ zIndex: `${zIndex}` }} id={gridKey} className={GridItemClasses} onClick={() => onSelect(title, gridKey)}>
                <div className="score">{score}</div>
                <div className="details">
                    <div className="title"><strong>{title}</strong></div>
                    <div className="by">By {by}</div>
                    <div>
                        Changes to our legacy lifecycle methods.
                        {
                            selected 
                            ? <span> In React 16.3.0, we are adding a few new lifecycle methods to assist with that migration. We are also introducing new APIs for long requested features: an official context API, a ref forwarding API, and an ergonomic ref API. 
                                <br/><br/>
                                <p>Reactâ€™s class component API has been around for years with little change. However, as we add support for more advanced features (such as error boundaries and the upcoming async rendering mode) we stretch this model in ways that it was not originally intended.</p>
                              </span> 
                            : null
                        }
                    </div>
                </div>
            </div>
        )
    }
}

GridItem.defaultProps = {
    selected: false,
    onSelect: () => {}
}

GridItem.propTypes = {
    by: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    score: PropTypes.number.isRequired,
    gridKey: PropTypes.string.isRequired,
    zIndex: PropTypes.number.isRequired,
    onSelect: PropTypes.func
}

export default GridItem;