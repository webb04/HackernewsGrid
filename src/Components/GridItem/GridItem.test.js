import React from 'react';
import ReactDOM from 'react-dom';
import GridItem from './GridItem';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<GridItem by='Fred' title="Title" gridKey="" zIndex={4} score={100}/>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
    const tree = renderer
      .create(<GridItem by='Fred' title="Title" gridKey="" zIndex={4} score={125}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Expands post', () => {
  jest.useFakeTimers()
  const Component = mount(<GridItem selected={true} by='Fred' title="Title" gridKey="" zIndex={4} score={125}/>);
  Component.setProps({ selected: true });
  expect(Component.find('.selected')).toHaveLength(1)

});

it('Closes post', () => {
  const Component = mount(<GridItem selected={true} by='Fred' title="Title" gridKey="" zIndex={4} score={125}/>);
  Component.setProps({ selected: false });
  expect(Component.find('.selected')).toHaveLength(0)
});