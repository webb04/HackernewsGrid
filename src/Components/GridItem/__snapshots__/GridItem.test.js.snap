// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  className="GridItem"
  id=""
  onClick={[Function]}
  style={
    Object {
      "zIndex": "4",
    }
  }
>
  <div
    className="score"
  >
    125
  </div>
  <div
    className="details"
  >
    <div
      className="title"
    >
      <strong>
        Title
      </strong>
    </div>
    <div
      className="by"
    >
      By 
      Fred
    </div>
    <div>
      Changes to our legacy lifecycle methods.
    </div>
  </div>
</div>
`;

exports[`renders correctly 1`] = `
<div
  className="GridItem"
  id=""
  onClick={[Function]}
  style={
    Object {
      "zIndex": "4",
    }
  }
>
  <div
    className="score"
  >
    125
  </div>
  <div
    className="details"
  >
    <div
      className="title"
    >
      <strong>
        Title
      </strong>
    </div>
    <div
      className="by"
    >
      By 
      Fred
    </div>
    <div>
      Changes to our legacy lifecycle methods.
    </div>
  </div>
</div>
`;
